### PR TITLE
convert android auto road label to a bitmap widget

### LINKED ISSUE #5794: [Android Auto] Integrate maps widgets for current road label
Similar to this issue https://github.com/mapbox/mapbox-navigation-android/issues/5793

Considering this ticket separate because there is a different framework for rendering the road label.

The `RoadLabelRenderer` will fit a road label text onto a Bitmap. For this ticket, instead of using the surfacelayer from android auto; try using the widget api. An example is here https://github.com/mapbox/mapbox-maps-android/tree/main/extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/widgets

Try deleting the surfacelayer module and then adding the functionality back with the new widget api.
https://github.com/mapbox/mapbox-navigation-android/tree/main/libnavui-androidauto/src/main/java/com/mapbox/androidauto/surfacelayer

### PR DESCRIPTION
**Pull request by Zayankovsky**
### Description
Closes #5794.

### CODE REVIEW

**Comment by kmadsen** [comment]
Thanks for finding and demonstrating this! Any BitmapWidget fix will happen in a `10.10.0` map version, and we're planning on releasing with support for `10.9.0`.

We can hold off on removing the `surfacelayer` approach and this will be a known issue that can be fixed with an upgrade.

File: libnavui-androidauto/src/main/java/com/mapbox/androidauto/navigation/roadlabel/RoadLabelSurfaceLayer.kt [hunk 1]
```diff
@@ -33,48 +32,40 @@ import com.mapbox.navigation.ui.shield.model.RouteShield
 @OptIn(MapboxExperimental::class)
 class RoadLabelSurfaceLayer(
     val carContext: CarContext,
-) : CarSurfaceLayer() {
```
Comment (kmadsen) [inline libnavui-androidauto/src/main/java/com/mapbox/androidauto/navigation/roadlabel/RoadLabelSurfaceLayer.kt:36]: in order to make this have a free upgrade, it seems we need to fix this part (in a separate pull request).

Comment (kmadsen) [reply libnavui-androidauto/src/main/java/com/mapbox/androidauto/navigation/roadlabel/RoadLabelSurfaceLayer.kt:36]: could just move this class to the internal package

Comment (kmadsen) [reply libnavui-androidauto/src/main/java/com/mapbox/androidauto/navigation/roadlabel/RoadLabelSurfaceLayer.kt:36]: but we also want to have a public component for this.

```
CurrentRoadLabel : MapboxCarMapObserver
```

**Comment by Zayankovsky** [comment]
> Thanks for finding and demonstrating this! Any BitmapWidget fix will happen in a `10.10.0` map version, and we're planning on releasing with support for `10.9.0`.
> 
> We can hold off on removing the `surfacelayer` approach and this will be a known issue that can be fixed with an upgrade.

Even if we don't migrate to `BitmapWidget` the issue still exists, because all other widgets disappear after you remove the road label.

File: libnavui-androidauto/src/main/java/com/mapbox/androidauto/navigation/roadlabel/RoadLabelSurfaceLayer.kt (hunk 1 above)
Comment (Zayankovsky) [reply libnavui-androidauto/src/main/java/com/mapbox/androidauto/navigation/roadlabel/RoadLabelSurfaceLayer.kt:36]: #6546

File: libnavui-androidauto/src/main/java/com/mapbox/androidauto/navigation/roadlabel/RoadLabelSurfaceLayer.kt [hunk 2]
```diff
@@ -137,5 +106,9 @@ internal class RoadLabelSurfaceLayerDelegate : CarSurfaceLayer() {
             .roundedLabelColor(Color.WHITE)
             .textColor(Color.BLACK)
             .build()
+
+        private val EMPTY_BITMAP = Bitmap.createBitmap(10, 10, Bitmap.Config.ARGB_8888).apply {
```
Comment (kmadsen) [inline libnavui-androidauto/src/main/java/com/mapbox/androidauto/navigation/roadlabel/RoadLabelSurfaceLayer.kt:110]:
```suggestion
        private val EMPTY_BITMAP = Bitmap.createBitmap(1, 1, Bitmap.Config.ARGB_8888).apply {
```

Size of this bitmap matter?

File: libnavui-androidauto/src/main/java/com/mapbox/androidauto/map/MapboxCarMapLoader.kt [hunk 3]
```diff
@@ -44,15 +47,21 @@ class MapboxCarMapLoader : MapboxCarMapObserver {
             logAndroidAuto("onAttached load style")
             mapSurface.getMapboxMap().loadStyle(
                 mapStyle(carContext.isDarkMode),
-                onStyleLoaded = {
+                onStyleLoaded = { style ->
                     logAndroidAuto("onAttached style loaded")
+                    style.addPersistentStyleCustomLayer(
```
Comment (kmadsen) [inline libnavui-androidauto/src/main/java/com/mapbox/androidauto/map/MapboxCarMapLoader.kt:52]: oh nice, perfect timing to have this map loader available 🪂

File: libnavui-androidauto/src/main/java/com/mapbox/androidauto/internal/surfacelayer/GLES20LayerHost.kt [hunk 4]
```diff
@@ -0,0 +1,32 @@
+package com.mapbox.androidauto.internal.surfacelayer
+
+import android.opengl.GLES20
+import com.mapbox.androidauto.internal.logAndroidAutoFailure
+import com.mapbox.maps.CustomLayerHost
+import com.mapbox.maps.CustomLayerRenderParameters
+
+// This layer fixes an issue on some Xiaomi devices that causes bitmap widgets not to render
```
Comment (kmadsen) [inline libnavui-androidauto/src/main/java/com/mapbox/androidauto/internal/surfacelayer/GLES20LayerHost.kt:8]: Nice find! 🥇

**Review by kmadsen — APPROVED** [review]
Good to see the CarTextLayouerHost removed in favor of an upstream api! 🎉

File: libnavui-androidauto/src/main/java/com/mapbox/androidauto/navigation/roadlabel/RoadLabelSurfaceLayer.kt (hunk 2 above)
Comment (Zayankovsky) [reply libnavui-androidauto/src/main/java/com/mapbox/androidauto/navigation/roadlabel/RoadLabelSurfaceLayer.kt]: No, this was simply copied from `CarTextLayerHost`.

**Comment by kmadsen** [comment]
Have been testing it on a Xiaomi POCO 3: MIUI Global 13.0.4

looks good

File: libnavui-androidauto/src/main/java/com/mapbox/androidauto/internal/surfacelayer/GLES20LayerHost.kt [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+class GLES20LayerHost : CustomLayerHost {
+
+    private var program = 0
+
+    override fun initialize() {
+        // Setup the OpenGL ES Program
+        program = GLES20.glCreateProgram()
```
Comment (kmadsen) [inline libnavui-androidauto/src/main/java/com/mapbox/androidauto/internal/surfacelayer/GLES20LayerHost.kt:15]: When testing, removing this does not make a difference. It works without creating a program

but it fails if you remove addPersistentStyleCustomLayer(GLES20LayerHost)

Comment (Zayankovsky) [reply libnavui-androidauto/src/main/java/com/mapbox/androidauto/internal/surfacelayer/GLES20LayerHost.kt]: Indeed, somehow I forgot to check that.

**Pull request merged**